test/suites/optional/multus.robot

```diff
@@ -0,0 +1,151 @@
+    ...    AND
+    ...    Interface Should Exist    ${PREEXISTING_BRIDGE_INTERFACE}
+    ...    AND
+    ...    Set IP For Host Interface    ${PREEXISTING_BRIDGE_INTERFACE}    ${PREEXISTING_BRIDGE_IP}
+
+    Curl Pod Over Local Interface    ${PREEXISTING_BRIDGE_POD_NAME}    ${NS}    ${PREEXISTING_BRIDGE_INTERFACE}
+
+    [Teardown]    Cleanup Bridge Test
+    ...    ${PREEXISTING_BRIDGE_NAD_YAML}
+    ...    ${PREEXISTING_BRIDGE_POD_YAML}
+    ...    ${PREEXISTING_BRIDGE_INTERFACE}
+
+No Pre-Existing Bridge Interface
+    [Documentation]    Test verifies if Bridge CNI plugin will
+    ...    work correctly if there is no pre-existing bridge
+    ...    interface - it needs to be created.
+    [Setup]    Run Keywords
+    ...    Interface Should Not Exist    ${BRIDGE_INTERFACE}
+    ...    AND
+    ...    Create NAD And Pod    ${BRIDGE_NAD_YAML}    ${BRIDGE_POD_YAML}
+    ...    AND
+    ...    Named Pod Should Be Ready    ${BRIDGE_POD_NAME}    ${NS}
+    ...    AND
+    ...    Interface Should Exist    ${BRIDGE_INTERFACE}
+    ...    AND
+    ...    Set IP For Host Interface    ${BRIDGE_INTERFACE}    ${BRIDGE_IP}
+
+    Curl Pod Over Local Interface    ${BRIDGE_POD_NAME}    ${NS}    ${BRIDGE_INTERFACE}
+
+    [Teardown]    Cleanup Bridge Test
+    ...    ${BRIDGE_NAD_YAML}
+    ...    ${BRIDGE_POD_YAML}
+    ...    ${BRIDGE_INTERFACE}
+
+
+*** Keywords ***
+Setup
+    [Documentation]    Test suite setup
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig
+
+Teardown
+    [Documentation]    Test suite teardown
+    Logout MicroShift Host
+    Remove Kubeconfig
+
+Create NAD And Pod
+    [Documentation]    Creates provided NetworkAttachmentDefinition and Pod.
+    [Arguments]    ${nad}    ${pod}
+    Oc Create    -f ${nad}
+    Oc Create    -f ${pod}
+
+Cleanup Bridge Test
+    [Documentation]    Removes provided NetworkAttachmentDefinition, Pod and network interface to allow for test rerun.
+    [Arguments]    ${nad}    ${pod}    ${if}
+    Run Keyword And Continue On Failure
+    ...    Oc Delete    -f ${pod}
+    Run Keyword And Continue On Failure
+    ...    Oc Delete    -f ${nad}
+    Command Should Work    ip link delete ${if}
+
+Curl Pod Over Local Interface
+    [Documentation]    Makes a HTTP request to 8080 for a given Pod over given interface.
+    [Arguments]    ${pod}    ${ns}    ${if}
+
+    ${networks}=    Get And Verify Pod Networks    ${pod}    ${ns}
+    ${extra_ip}=    Set Variable    ${networks}[1][ips][0]
+
+    ${stdout}=    Command Should Work    curl -v --interface ${if} ${extra_ip}:8080
+    Should Contain    ${stdout}    Hello MicroShift
+
+Interface Should Not Exist
+    [Documentation]    Verifies that network interface does not exist.
+    [Arguments]    ${if}
+    ${stdout}    ${rc}=    Execute Command
+    ...    ip link show ${if}
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    ${rc}    1
```

Comment:
Feels as if we would benefit from `Command Should Not Work` keyword.

Reply:
Done